### DISCUSSION TITLE
JWT cookieName option does not work

### DISCUSSION THREAD

**Discussion by tyrw** [nextauthjs/next-auth#1882] [Help] [answered]
## Description 🐜

The `cookieName` option for the `jwt` section as described in [the docs](https://next-auth.js.org/configuration/options#jwt) does not appear to work.  The docs state:

> It also supports the following options:
> - `secureCookie` - (boolean) Use secure prefixed cookie name
> By default, the helper function will attempt to determine if it should use the secure prefixed cookie (e.g. `true` in production and `false` in development, unless NEXTAUTH_URL contains an HTTPS URL).
> - `cookieName` - (string) Session token cookie name
> The secureCookie option is ignored if `cookieName` is explicitly specified.

## How to reproduce ☕️

Clone the example repo

```sh
git clone https://github.com/nextauthjs/next-auth-example.git
cd next-auth-example
npm i
```

Add the `cookieName` option:

```js
// In /pages/api/auth/[...nextauth].js

{
  jwt: {
    cookieName: "custom-name"
  }
}
```

This is not used by the application.  It still uses `next-auth.session-token`.

In the source code for `jwt.js`, it does not appear to respect the setting:
[https://github.com/nextauthjs/next-auth/blob/543f812eb32448044d2aa725b623ca1dedbb68a3/src/lib/jwt.js#L115-L117](https://github.com/nextauthjs/next-auth/blob/543f812eb32448044d2aa725b623ca1dedbb68a3/src/lib/jwt.js#L115-L117)

## Screenshots / Logs 📽

Chrome tools shows that the cookie name remains `next-auth.session-token`.

## Contributing 🙌🏽

Potentially interested in contributing, assuming I have understood the `jwt.js` file correctly.

**Comment by balazsorban44** [comment]
Did some digging, and it looks like this is the reminiscent of version 2.x https://github.com/nextauthjs/next-auth/commit/416785941b5e2efa0da91e4077e11cca4dcc5c40#diff-2c2a1625654e55c4a251b587dc0196c4aba9c610825e56ce2b861762888c7155R29

In that case, the docs (And typescript https://github.com/nextauthjs/next-auth/blob/0eb4159737e66a15e2d5f2d9c3236117226d9baf/types/jwt.d.ts#L49) should be updated

**Comment by tyrw** [comment]
> should be updated

Are you looking for an update to remove the option from the docs, or to re-implement the feature?  I'm not clear on why it was removed in the first place.

**Comment by balazsorban44** [comment]
Update the docs. To be honest, I don't know either, @iaincollins did it a long time ago, I wasn't around then yet, sorry.

I think this is the particular diff that changed it: 
https://github.com/nextauthjs/next-auth/commit/86921022dc7f0b4fcf8897e6ee60be0ae1aaa99f#diff-2c2a1625654e55c4a251b587dc0196c4aba9c610825e56ce2b861762888c7155L28-L30

**Comment by balazsorban44** [comment]
Wait, sorry!

https://github.com/nextauthjs/next-auth/blob/0eb4159737e66a15e2d5f2d9c3236117226d9baf/src/lib/jwt.js#L107-L124

The current code only implements a default, but if you provide a cookieName, it SHOULD be used after all :facepalm:

**Comment by balazsorban44** [comment, marked as answer]
using `getToken({cookieName: "some-name"})` do seem to work for me! Do you have a reproduction?

This 
![image](https://user-images.githubusercontent.com/18369201/116484204-e111f080-a888-11eb-88fd-f490a2559815.png) seem to be in the incorrect place.

Check the docs again: https://next-auth.js.org/configuration/options#jwt-helper

It is a parameter for the `getToken()` method.

You may also have to change it here: https://next-auth.js.org/configuration/options#cookies

Do note that this can introduce security issues. Can you maybe explain why do you need to change the name of this cookie?

**Comment by tyrw** [comment]
Here is a reproduction of it not working:

https://replit.com/@userfront/next-auth-example#pages/api/auth/[...nextauth].js

You can see when you log in (only Google is enabled) that the cookie is set to the default name.

**Reply by balazsorban44** [reply]
Please read my above comment carefully again. 😊 You set the cookieName in the wrong place.

**Reply by tyrw** [reply]
Thanks, I was just about to comment!  It does work, though I would consider it to be a bug in the API, so to speak, that `cookieName` does not set the cookie name.  Thanks for your help!

**Reply by balazsorban44** [reply]
Again, please read carefully! You have to set it in a different place. What you implemented is not documented anywhere in our documentation pages nor does it appear in our codebase.

The cookieName is something you pass to the getToken function, and/or the cookies advanced option. I linked it above.

**Reply by tyrw** [reply]
Yes I found it, thank you. My comment was simply that setting the cookie name using sessionToken instead of cookieName is confusing to a newcomer.
